src/ConnectionPool.jl

```diff
@@ -70,17 +69,19 @@ mutable struct Connection{T <: IO}
     peerport::UInt16 # debug only
     localport::UInt16 # debug only
     io::T
+    clientconnection::Bool
     buffer::IOBuffer
     sequence::Int
     writecount::Int
-    writebusy::Bool
-    writedone::Condition
+    writelock::Channel{Bool}
     readcount::Int
-    readbusy::Bool
-    readdone::Condition
+    readlock::Channel{Bool}
     timestamp::Float64
 end
 
+hashconn(x::Connection{T}) where {T} = hashconn(T, x.host, x.port, x.pipeline_limit, x.require_ssl_verification, x.clientconnection)
+hashconn(T, host, port, pipeline_limit, require_ssl_verification, client) = hash(T, hash(host, hash(port, hash(pipeline_limit, hash(require_ssl_verification, hash(client, UInt(0)))))))
```

Comment:
Used for connection pooling Dict key; these properties "make up" a type of connection that can be re-used by another request that also has the same properties.

Reply:
That makes sense - could do with just putting your explanation into the code as a comment?

Reply:
Though for the reasons discussed further down it may not be helpful to do the hash here, unless the lookup is insensitive to hash collisions.  Could be renamed `connection_reuse_key`?